src/options/view.rs

```diff
@@ -169,13 +181,28 @@ impl details::Options {
 
         Ok(details::Options {
             table: Some(TableOptions::deduce(matches, vars)?),
-            header: matches.has(&flags::HEADER)?,
+            header: Self::deduce_header(matches, stdout_istty)?,
             xattr: xattr::ENABLED && matches.has(&flags::EXTENDED)?,
             secattr: xattr::ENABLED && matches.has(&flags::SECURITY_CONTEXT)?,
             mounts: matches.has(&flags::MOUNTS)?,
             color_scale: ColorScaleOptions::deduce(matches, vars)?,
         })
     }
+
+    fn deduce_header(matches: &MatchedFlags<'_>, stdout_istty: bool) -> Result<bool, OptionsError> {
+        let word = if let Some(w) = matches.get(&flags::HEADER)? {
+            w.to_os_string()
+        } else {
+            return Ok(false);
+        };
+
+        match word.to_string_lossy().as_ref() {
```

Comment:
this is allocating unnecessarily, its calling `to_os_string()` and then again with `to_string_lossy()`, can we try something like this?
```rust
use crate::options::flags::WHEN;
```
```rust
    fn deduce_header(matches: &MatchedFlags<'_>, stdout_istty: bool) -> Result<bool, OptionsError> {
        match matches.get(&flags::HEADER)? {
            Some(word) if word == WHEN[0] => Ok(true), // always
            Some(word) if word == WHEN[1] => Ok(stdout_istty), // auto
            Some(word) if word == WHEN[2] => Ok(false), // never
            Some(other) => Err(OptionsError::BadArgument(&flags::HEADER, other.into())), // bad arg
            None => Ok(false), // no flag
        }
    }
```